test/unit/middleware/resolve_service_test.js

```diff
@@ -8,24 +8,88 @@ const {expect} = require('chai')
 const resolveService = require('../../../app/middleware/resolve_service')
 const userFixtures = require('../../fixtures/user_fixtures')
 
-describe('resolve service', () => {
-  it('resolve service from path param', () => {
-    const res = {render: sinon.spy()}
-    const nextSpy = sinon.spy()
+describe('resolve service middleware', () => {
+  let res, nextSpy, user
 
-    const user = userFixtures.validUser().getAsObject()
+  beforeEach(() => {
+    res = { setHeader: sinon.spy(), status: sinon.spy(), render: sinon.spy() }
+    nextSpy = sinon.spy()
+    user = userFixtures.validUser().getAsObject()
+  })
+
+  it('from externalServiceId in path param then remove it', () => {
     const req = {user: user, params: {externalServiceId: user.serviceRoles[0].service.externalId}}
 
     resolveService(req, res, nextSpy)
 
     expect(req.service).to.deep.equal(user.serviceRoles[0].service)
+    expect(req.params.externalServiceId).to.equal(undefined) // Removing becuase we going forward we only want to resolve the service via req.service as this way we know the person authorised
     expect(nextSpy.called).to.equal(true)
   })
 
-  it('service.hasDirectDebitGatewayAccount is true and service.hasCardGatewayAccount is false when we have Direct Debit gateway accounts only', () => {
-    const res = {render: sinon.spy()}
-    const nextSpy = sinon.spy()
+  it('from externalServiceId in path param but show unauthorised view if it does not belong to this user', () => {
+    const req = {user: user, params: {externalServiceId: 'someoneElsesID'}}
+
+    resolveService(req, res, nextSpy)
+
+    expect(req.service).to.equal(undefined)
+    expect(res.status.calledWith(403))
+    expect(res.render.calledWith('error'))
+    expect(nextSpy.called).to.equal(false)
+  })
+
+  it('from gatewayAccountId in cookie', () => {
+    const req = {
+      user: user,
+      params: {},
+      gateway_account: {
+        currentGatewayAccountId: user.serviceRoles[0].service.externalServiceId
```

Comment:
don't think this set up is quite right? why does a gateway account id = a service id?